include/rcpputils/shared_library.hpp

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCPPUTILS__SHARED_LIBRARY_HPP_
+#define RCPPUTILS__SHARED_LIBRARY_HPP_
+
+#include <string>
+#include <stdexcept>
+
+#include "rcutils/shared_library.h"
+#include "rcpputils/visibility_control.hpp"
+
+namespace rcpputils
+{
+
+/**
+ * This class is an abstraction of rcutils shared library to be able to used it
+ *  with modern C++.
+ **/
+class RCPPUTILS_PUBLIC SharedLibrary
+{
+public:
+  /// The library is loaded in the constructor.
+  /**
+   * \param[in] input The library string path.
+   * \throws std::bad_alloc if allocating storage for the callback fails
+   * \throws std::runtime_error if there are some invalid arguments or the library
+   * was not load properly
+   */
+  RCPPUTILS_PUBLIC
+  explicit SharedLibrary(const std::string & library_path);
+
+  /// The library is unloaded in the deconstructor
+  RCPPUTILS_PUBLIC
+  virtual ~SharedLibrary();
```

Comment:
Why is (only) the destructor virtual?

Reply:
I think they're just trying to follow C++ best practices, which is "all destructors should be virtual". Even if this is not intended to be subclassed, somebody could do so, and it should clean itself up, which will not happen if the dtor is not virtual.

Reply:
Indeed, if you wanted to make it non virtual then you could mark the class final or make the destructor protected so no one could try to delete a base class pointer to this.